uint8 counter_value
